apiv2/retention/retention.go

```diff
@@ -0,0 +1,219 @@
+package retention
+
+import (
+	"context"
+	"errors"
+	"strconv"
+
+	"github.com/go-openapi/runtime"
+	v2client "github.com/mittwald/goharbor-client/apiv2/internal/api/client"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client/products"
+	model "github.com/mittwald/goharbor-client/apiv2/model/legacy"
+	"github.com/mittwald/goharbor-client/apiv2/project"
+	pc "github.com/mittwald/goharbor-client/apiv2/project"
+)
+
+const (
+	AlgorithmOr string = "or"
+
+	// Key for defining matching repositories
+	ScopeSelectorRepoMatches ScopeSelector = "repoMatches"
+
+	// Key for defining excluded repositories
+	ScopeSelectorRepoExcludes ScopeSelector = "repoExcludes"
+
+	// Key for defining matching tag expressions
+	TagSelectorMatches TagSelector = "matches"
+
+	// Key for defining excluded tag expressions
+	TagSelectorExcludes TagSelector = "excludes"
+
+	// The kind of the retention selector, _always_ defaults to 'doublestar'
+	SelectorTypeDefault string = "doublestar"
+
+	// Retain the most recently pushed n artifacts - count
+	PolicyTemplateLatestPushedArtifacts PolicyTemplate = "latestPushedK"
+
+	// Retain the most recently pulled n artifacts - count
+	PolicyTemplateLatestPulledArtifacts PolicyTemplate = "latestPulledN"
+
+	// Retain the artifacts pushed within the last n days
+	PolicyTemplateDaysSinceLastPush PolicyTemplate = "nDaysSinceLastPush"
+
+	// Retain the artifacts pulled within the last n days
+	PolicyTemplateDaysSinceLastPull PolicyTemplate = "nDaysSinceLastPull"
+
+	// Retain always
+	PolicyTemplateRetainAlways PolicyTemplate = "always"
+)
+
+type Client interface {
+	NewRetentionPolicy(ctx context.Context, scopeSelector ScopeSelector, projectRef int64,
+		retentionPolicyTemplate PolicyTemplate, tagSelector TagSelector, retentionRuleParam map[PolicyTemplate]interface{},
+		repoPattern, tagPattern, cronSchedule string, untaggedArtifacts bool) error
+	GetRetentionPolicyByProjectID(ctx context.Context, projectID int64) (*model.RetentionPolicy, error)
```

Comment:
Instead of `projectID int64` as an argument, you should use `project *model.Project`. This is consistent with the api design and what most other methods use too.